### PR TITLE
DOCS: Updated docblocks of file link methods

### PR DESCRIPTION
**Pull request by sminnee**
... to clarify expectations about return values

This is a response to this discussion about public APIs https://github.com/silverstripe/silverstripe-framework/pull/4229#discussion_r31983112 

We should be clearer about our edge-case expectations of APIs. I'd like to see this being a lot more complete for 4.0 if possible (a man can dream!) :-)

### CODE REVIEW

**Comment by dhensby** [comment]
@sminnee Seems fine, just curious as to why you've effectively swapped the docs and marked the original functions as aliases?

**Comment by tractorcow** [comment]
I think @dhensby might be right.

**Comment by sminnee** [comment]
It was to set expectations about which one was preferred. Should I rebase this?

File: filesystem/File.php [hunk 1]
```diff
@@ -249,7 +249,9 @@ public static function find($filename) {
 	}
 
 	/**
-	 * Just an alias function to keep a consistent API with SiteTree
+	 * Returns a web link download/view to the file. May be absolute or relateive, depending on configuration.
```
Comment (tractorcow) [inline filesystem/File.php]: `relative`

**Comment by tractorcow** [comment]
I think it's probably fine, but I think it's more usual for the alias to invoke the method it's aliasing, rather than the other way around. :P Not a merge blocker really.

Please rebase and I'll merge. :) You may as well fix that spelling mistake while you're at it.

**Comment by tractorcow** [comment]
I think this PR is well out of date; In any case all 4.0 File methods have been rewritten.

**Pull request closed without merging**